staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -167,6 +205,11 @@ func (q *Type) Add(item interface{}) {
 		return
 	}
 	if q.dirty.has(item) {
+		// the same item is added again before it is processed, call the Touch
+		// function if the queue cares about it (for e.g, reset its priority)
+		if !q.processing.has(item) {
+			q.queue.Touch(item)
+		}
```

Comment:
this is the non-trivial change in the PR. Shouldn't there be tests?

Reply:
I think there should be.

I plan to add another dummy Queue implementation in the test package which traces whether `Touch` function is called and then uses the new Queue implementation in:

* TestCollapse (test that `Touch` function is indeed called)
* TestCollapseWhileProcessing (test that `Touch` function is not called)

Does this sound like a good plan?

Reply:
Tests updated.